onnx/defs/schema.cc

```diff
@@ -705,26 +730,45 @@ void OpSchema::ParseAndSetTypes(
   }
 }
 
-OpSchema& OpSchema::SetContextDependentFunctionBodyBuilder(ContextDependentFunctionBodyBuilder functionBuilder) {
+OpSchema& OpSchema::SetContextDependentFunctionBodyBuilder(
+    ContextDependentFunctionBodyBuilder functionBuilder) {
   functionBuilder_ = functionBuilder;
   return *this;
 }
 
-bool OpSchema::BuildContextDependentFunction(const FunctionBodyBuildContext& ctx, FunctionProto& functionProto) const {
+bool OpSchema::BuildContextDependentFunction(
+    const FunctionBodyBuildContext& ctx,
+    FunctionProto& functionProto) const {
   if (functionBuilder_)
     return functionBuilder_(ctx, *this, functionProto);
   else
     return false;
 }
 
 OpSchema& OpSchema::FunctionBody(const std::vector<NodeProto>& func_nodes) {
+  // By default, the function body graph is relying on the OperatorSet this
+  // function belongs to.
+  auto relied_opset = function_body_.mutable_opset_import()->Add();
+  relied_opset->set_domain(this->domain());
+  relied_opset->set_version(this->SinceVersion());
+
   for (const auto node : func_nodes) {
     auto new_node = function_body_.add_node();
     new_node->CopyFrom(node);
   }
   return *this;
 }
 
+OpSchema& OpSchema::FunctionBody(
+    const std::vector<NodeProto>& func_nodes,
+    const std::vector<OperatorSetIdProto>& relied_opsets) {
+  for (auto& relied_opset : relied_opsets) {
```

Comment:
I think it would be a good idea to include the function's domain/since-version in this case also (or, at least if it is not a duplicate).

Reply:
it's included, it's call the other overloaded FunctionBody, which adds that.